src/PostgREST/ApiRequest.hs

```diff
@@ -92,23 +90,17 @@ data Resource
   | ResourceRoutine Text
   | ResourceSchema
 
-data ActionRelation
-  = ActRead Bool
-  | ActMutate Mutation
-  | ActRelInfo
-
-data ActionRoutine
-  = ActInvoke InvokeMethod
-  | ActRoutInfo
-
-data ActionSchema
-  = ActSchemaRead Bool
-  | ActSchemaInfo
+data DbAction
+  = ActRelationRead {dbActQi :: QualifiedIdentifier, actHeadersOnly :: Bool}
+  | ActRelationMut  {dbActQi :: QualifiedIdentifier, actMutation :: Mutation}
+  | ActRoutine      {dbActQi :: QualifiedIdentifier, actInvMethod :: InvokeMethod}
 
 data Action
-  = ActRelation QualifiedIdentifier ActionRelation
-  | ActRoutine QualifiedIdentifier ActionRoutine
-  | ActSchema Schema ActionSchema
+  = ActDb           DbAction
+  | ActSchemaRead   Schema Bool
+  | ActRelationInfo QualifiedIdentifier
+  | ActRoutineInfo  QualifiedIdentifier
+  | ActSchemaInfo   Schema
```

Comment:
```suggestion
  | ActSchemaInfo
```

I think the schema is not needed anymore here.

Reply:
Will address that in a new PR